tests/store/tracking/test_file_store.py

```diff
@@ -331,6 +331,26 @@ def test_get_experiment_int_experiment_id_backcompat(self):
         with safe_edit_yaml(root_dir, "meta.yaml", self._experiment_id_edit_func):
             self._verify_experiment(fs, exp_id)
 
+    def test_get_experiment_retries_for_transient_empty_yaml_read(self):
+        fs = FileStore(self.test_root)
+        exp_name = random_str()
+        exp_id = fs.create_experiment(exp_name)
+
+        mock_empty_call_count = 0
+
+        def mock_read_yaml(*args, **kwargs):
+            nonlocal mock_empty_call_count
+            if mock_empty_call_count < 2:
+                mock_empty_call_count += 1
+                return None
+            else:
+                return read_yaml(*args, **kwargs)
+
+        with mock.patch("mlflow.store.tracking.file_store.read_yaml", mock_read_yaml):
+            fetched_experiment = fs.get_experiment(exp_id)
```

Comment:
```suggestion
            fetched_experiment = fs.get_experiment(exp_id)
            assert mock_empty_call_count == 2
```

to ensure `mock_read_yaml` is used.

Reply:
```python
        with mock.patch(
            "mlflow.store.tracking.file_store.read_yaml",
            side_effect=[None, None, read_yaml(os.path.join(self.test_root, exp_id), "meta.yaml")],
        ) as mock_read_yaml:
            fetched_experiment = fs.get_experiment(exp_id)
            assert mock_read_yaml.call_count == 3
            assert fetched_experiment.experiment_id == exp_id
            assert fetched_experiment.name == exp_name
```

also works.

Reply:
Added a call count assert :).